### PR TITLE
HSH-190: Run Hail locally

### PR DESCRIPTION
**Pull request by GennadiyZakharov**
Changes hl_init() to spin up Hail both on a cluster and on the local machine.
All pipeline scripts use init_hl() to spin up Hail

### CODE REVIEW

File: wes_qc/hail_utils.py [hunk 1]
```diff
@@ -18,19 +20,22 @@ def clear_temp_folder(tmp_dir: str) -> None:
...
+
     print("=== Initializing Hail ===")
-    hl.init(sc=sc, tmp_dir=tmp_dir, idempotent=True)
+    hl.init(tmp_dir=tmp_dir, idempotent=True)
     hl.default_reference("GRCh38")
-    return sc
 
 
-def stop_hl(sc: pyspark.SparkContext) -> None:
+def stop_hl(sc: Optional[pyspark.SparkContext] = None) -> None:
```
Comment (Eric-Kobayashi) [inline wes_qc/hail_utils.py:38]: this is probably unnecessary

File: wes_qc/hail_utils.py [hunk 2]
```diff
@@ -18,19 +20,22 @@ def clear_temp_folder(tmp_dir: str) -> None:
     os.makedirs(tmp_dir, exist_ok=True)
 
 
-def init_hl(tmp_dir: str) -> pyspark.SparkContext:
+def init_hl(tmp_dir: str) -> Optional[pyspark.SparkContext]:
```
Comment (Eric-Kobayashi) [inline wes_qc/hail_utils.py:23]: it is returning None

File: 3-variant_qc/7-plot_rf_output.py [hunk 3]
```diff
@@ -592,6 +594,7 @@ def main():
     plot_dir = os.path.join(root_plot_dir, "variant_qc", model_id)
 
     # = STEP LOGIC = #
+    hail_utils.init_hl(tmp_dir)
```
Comment (Eric-Kobayashi) [inline 3-variant_qc/7-plot_rf_output.py:597]: why adding it here?

**Review by Eric-Kobayashi — CHANGES_REQUESTED** [review]

File: 3-variant_qc/7-plot_rf_output.py (hunk 3 above)
Comment (GennadiyZakharov) [reply 3-variant_qc/7-plot_rf_output.py:597]: The function create_plots() reads Hail tables, and it triggers the implicit init of Hail. I decided that it's better to init Hail explicitely.

File: wes_qc/hail_utils.py (hunk 2 above)
Comment (GennadiyZakharov) [reply wes_qc/hail_utils.py:23]: Fixed.

File: wes_qc/hail_utils.py (hunk 1 above)
Comment (GennadiyZakharov) [reply wes_qc/hail_utils.py:38]: Commented this for now. After a while, when we are sure that Hail init works correctly in all cases, we can delete it.

**Pull request merged**